test/options.js

```diff
@@ -0,0 +1,66 @@
+const test = require('brittle')
+const ffmpeg = require('..')
+
+const { options } = ffmpeg
+
+test('getOption', (t) => {
+  const ctx = new ffmpeg.CodecContext(ffmpeg.Codec.AV1.encoder)
+  const result = options.getOption(ctx._handle, 'threads')
+  t.is(result, '1')
+})
+
+test('setOption', (t) => {
+  const ctx = new ffmpeg.CodecContext(ffmpeg.Codec.AV1.encoder)
+  options.setOption(ctx._handle, 'threads', '2')
+  const result = options.getOption(ctx._handle, 'threads')
+  t.is(result, '2')
+})
+
+test('listOptionNames', (t) => {
+  const ctx = new ffmpeg.CodecContext(ffmpeg.Codec.AV1.encoder)
+  const result = options.listOptionNames(ctx._handle)
+  t.ok(result && result.length)
+})
```

Comment:
I wonder if could not use the `options` API as internals of classes which have options.

Then instead we could have:

```js
  const ctx = new ffmpeg.CodecContext(ffmpeg.Codec.AV1.encoder)
  const result = ctx.listOptionNames()
```

We could have an `FFMpegOptions` class that our classes inherits from (we could probably use composition instead)

---------

That would not change that unit test tho, even if we had a class  `FFMpegOptions`, it should be tested unitary.

Reply:
Also, the final user should not rely on internal fields like `_handle`. Even with the current approach we should have at least:

```suggestion
test('listOptionNames', (t) => {
  const ctx = new ffmpeg.CodecContext(ffmpeg.Codec.AV1.encoder)
  const result = options.listOptionNames(ctx)
  t.ok(result && result.length)
})
```

Reply:
Okay pretty neat then :)